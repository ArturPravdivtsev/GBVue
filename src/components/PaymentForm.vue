<template>
  <div>
    <v-dialog
      ref="dialog"
      v-model="modal"
      :return-value.sync="date"
      persistent
      width="290px"
    >
      <template v-slot:activator="{ on }">
        <v-text-field
          v-model="date"
          label="Date"
          readonly
          v-on="on"
        ></v-text-field>
      </template>
      <v-date-picker v-model="date" scrollable>
        <v-spacer></v-spacer>
        <v-btn text color="primary" @click="modal = false"> Cancel </v-btn>
        <v-btn text color="primary" @click="$refs.dialog.save(date)">
          OK
        </v-btn>
      </v-date-picker>
    </v-dialog>
    <v-text-field v-model="category" label="Category" required></v-text-field>
    <v-text-field v-model="value" label="Value" required></v-text-field>

    <v-btn color="primary" @click="save">Save</v-btn>
  </div>
</template>

<script>
import { mapMutations, mapGetters } from "vuex";
export default {
  data() {
    return {
      date: "",
      category: "",
      value: 0,
      modal: false,
    };
  },
  props: {
    id: Number,
  },
  methods: {
    ...mapMutations(["addData", "editData"]),
    save() {
      const { date, category, value, id } = this;
      if (this.id) {
        this.editData({ date, category, value, id });
      } else {
        this.addData({ date, category, value });
      }
    },
    formatDate(date) {
      var dd = date.getDate();
      if (dd < 10) dd = "0" + dd;
      var mm = date.getMonth() + 1;
      if (mm < 10) mm = "0" + mm;
      var yyyy = date.getFullYear();
      return dd + "." + mm + "." + yyyy;
    },
  },
  computed: {
    ...mapGetters(["getPaymentsList"]),
  },
  mounted() {
    if (this.$route.params.category) {
      this.category = this.$route.params.category;
      this.date = this.formatDate(new Date());
      if (this.$route.query.value) {
        this.value = this.$route.query.value;
        this.save();
      }
    }
    if (this.id) {
      const item = this.getPaymentsList.find((item) => item.id === this.id);
      if (item) {
        this.date = item.date;
        this.category = item.category;
        this.value = item.value;
      }
    }
  },
};
</script>

<style>
</style>