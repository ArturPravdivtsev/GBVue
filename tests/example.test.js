import 'regenerator-runtime/runtime'
import PaymentForm from '../src/components/PaymentForm'
import { mount, createLocalVue, shallowMount } from '@vue/test-utils'
import Vuex from 'vuex'
import Vuetify from 'vuetify'
import VueRouter from 'vue-router'

const localVue = createLocalVue()
localVue.use(Vuex)
localVue.use(VueRouter)

const router = new VueRouter()

describe('PaymentForm', () => {
    let actions
    let mutations
    let store
    let vuetify

    beforeEach(() => {
        actions = {
            save: jest.fn()
        }
        mutations = {
            addData: jest.fn()
        }

        store = new Vuex.Store({
            actions,
            mutations
        })

        vuetify = new Vuetify()
    })
    test('PaymentForm renders a vue instance', () => {
        expect(
            shallowMount(PaymentForm, {
                    localVue,
                    router,
                    vuetify,
                    store
                }
            ).isVueInstance()
        ).toBe(true);
    })
    test('PaymentForm not changed', () => {
        const Wrapper = shallowMount(PaymentForm, {
            localVue,
            router,
            vuetify,
            store
          })
        expect(Wrapper.html()).toMatchSnapshot()
    })

    test('PaymentForm trigger Date Picker', () => {
        const Wrapper = shallowMount(PaymentForm, {
            localVue,
            router,
            vuetify,
            store
        })
        Wrapper.find('v-text-field').trigger('click')
        Wrapper.vm.$nextTick()
        expect(Wrapper.find('v-date-picker[name="date-picker"]').isVisible()).toBe(true)
    })

    test('PaymentForm save', () => {
        const Wrapper = shallowMount(PaymentForm, {
            store,
            localVue,
            router,
            vuetify,
            
        })
        Wrapper.setData({ date: '18-03-2022', category: 'Educations', value: '1000' })
        Wrapper.find('v-btn').trigger('click')
        Wrapper.vm.$emit('save');
        expect(Wrapper.emitted().save).toBeTruthy()
    })

    test('PaymentForm right data', () => {
        const Wrapper = shallowMount(PaymentForm, {
            store,
            localVue,
            router,
            vuetify,
            
        })
        const date = Wrapper.find('v-text-field[name="date-activator"]')
        date.setValue('18-03-2022')
        const category = Wrapper.find('v-text-field[name="category"]')
        category.setValue('Education')
        const value = Wrapper.find('v-text-field[name="value"]')
        value.setValue('1000')

        expect(Wrapper.vm.date).toBe('18-03-2022')
        expect(Wrapper.vm.category).toBe('Education')
        expect(Wrapper.vm.value).toBe('1000')
    })
})