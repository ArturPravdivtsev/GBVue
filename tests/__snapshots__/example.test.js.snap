// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentForm PaymentForm not changed 1`] = `
"<div>
  <v-dialog return-value=\\"\\" persistent=\\"\\" width=\\"290px\\" name=\\"dialog\\">
    <v-date-picker scrollable=\\"\\" name=\\"date-picker\\">
      <v-spacer></v-spacer>
      <v-btn text=\\"\\" color=\\"primary\\" name=\\"date-close\\">
        Cancel
      </v-btn>
      <v-btn text=\\"\\" color=\\"primary\\">
        OK
      </v-btn>
    </v-date-picker>
  </v-dialog>
  <v-text-field label=\\"Category\\" required=\\"\\" name=\\"category\\"></v-text-field>
  <v-text-field label=\\"Value\\" required=\\"\\" name=\\"value\\"></v-text-field>
  <v-btn color=\\"primary\\" name=\\"save\\">Save</v-btn>
</div>"
`;
